src/Syntax/TextmateTokenizerJob.re

```diff
@@ -115,38 +130,52 @@ let doWork = (pending: pendingWork, completed: completedWork) => {
 
     Log.tracef(m => m("Tokenizing line: %i", currentLine));
 
+    let line = pending.lines[currentLine] ++ "\n";
+
     // Get new tokens & scopes
     let (tokens, scopes) =
       Textmate.Tokenizer.tokenize(
         ~lineNumber=currentLine,
         ~scopeStack=scopes,
         ~scope=pending.scope,
         pending.tokenizer,
-        pending.lines[currentLine] ++ "\n",
+        line,
       );
 
+    let isWhitespaceOnly = (startIndex, endIndex) =>
+      try(
+        {
+          for (idx in startIndex to endIndex - 1) {
+            if (!StringEx.isSpace(line.[idx])) {
+              raise(NoWhitespaceException);
+            };
+          };
+          true;
+        }
+      ) {
+      | NoWhitespaceException => false
+      };
```

Comment:
```suggestion
    let rec isWhitespaceOnly = (startIndex, endIndex) =>
      if (startIndex >= endIndex) {
        true;
      } else {
        !StringEx.isSpace(line.[startIndex] && isWhitespaceOnly(startIndex + 1, endIndex);
      };
```

There's a pattern here. Opportunity for `StringEx.forAll` with an optional start and end index?

Reply:
Added a `StringEx.forAll` to help with this